### DISCUSSION TITLE
What is the width-height ratio for which icons are rendered in 2 cells instead of one?

### DISCUSSION THREAD

**Discussion by ymn64** [kovidgoyal/kitty#6669] [Q&A] [answered]
I am designing an icon pack and I need to determine this ratio threshold to ensure all icons render well in 2 spaces.
Also while we're at it, is there a way (some hacky configuration or something) to make sure this works for icons with varying ratios, as adjusting their ratio can really distort their original design?

**Comment by kovidgoyal** [comment]
There is no threshold. Widths of characters are set by the unicode
standard: https://sw.kovidgoyal.net/kitty/faq/#some-special-symbols-are-rendered-small-truncated-in-kitty

For PUA characters when a char is followed by a space kitty will use
that extra space, if the glyph rendered at the current font size is
wider than a single cell.

**Reply by ymn64** [reply]
Yeah I'm talking specifically about PUA characters. Got it, so this dynamically decided by Kitty.

Are you aware of a hack to trick Kitty to think all my glyphs need 2 spaces to render? I mean something I can do upstream with my FontForge configuration, currently I simply set `glyph.width = constant` for all the glyphs (but even when I change the width for a specific glyph, nothing changes.)

Btw, I don't have this issue with iTerm and Wezterm.

**Reply by kovidgoyal** [reply, marked as answer]
No, the width of characters in a terminal has to be fixed otherwise you would have chaos since terminal programs would not know what width a character has. THerefore all PUA characters must be width 1. THe only exception is if they happen to be followed by a space, then kitty uses that space to *render* the character but the cursor will still move only once cell and the character will still be technicall even if not visually in one cell. There is nothing you can do in a font to influence that. If you stop and think about it that  must be so because terminal programs can have no idea what font a terminal emulator is using. THerefore widths cannot depend on fonts.

**Reply by ymn64** [reply]
Got it, thank you.
